src/js/components/DataFilters/DataFilters.js

```diff
@@ -0,0 +1,136 @@
+import React, { Children, useContext, useState } from 'react';
+import { Close } from 'grommet-icons/icons/Close';
+import { Filter } from 'grommet-icons/icons/Filter';
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { DataFilter } from '../DataFilter';
+import { DataForm } from '../Data';
+import { DropButton } from '../DropButton';
+import { Header } from '../Header';
+import { Heading } from '../Heading';
+import { Layer } from '../Layer';
+import { DataContext } from '../../contexts/DataContext';
+import { MessageContext } from '../../contexts/MessageContext';
+import { DataFiltersPropTypes } from './propTypes';
+
+const dropProps = {
+  align: { top: 'bottom', right: 'right' },
+};
+
+export const DataFilters = ({ drop, children, heading, layer, ...rest }) => {
+  const { clearFilters, data, messages, properties, view } =
+    useContext(DataContext);
+  const { format } = useContext(MessageContext);
+  const [showContent, setShowContent] = useState();
+  const controlled = drop || layer;
+
+  const clearControl = clearFilters && (
+    <Box flex={false}>
+      <Button
+        label={format({
+          id: 'dataFilters.clear',
+          messages: messages?.dataFilters,
+        })}
+        onClick={clearFilters}
+      />
+    </Box>
+  );
+
+  let filters;
+  if (Children.count(children) === 0) {
+    let filtersFor;
+    if (!properties && data && data.length) filtersFor = Object.keys(data[0]);
+    else if (Array.isArray(properties)) filtersFor = properties;
+    else if (typeof properties === 'object')
+      filtersFor = Object.keys(properties);
+    else filtersFor = [];
+    filters = filtersFor.map((property) => (
+      <DataFilter key={property} property={property} />
+    ));
+  }
+
+  const content = (
+    <DataForm
+      pad={controlled ? 'medium' : undefined}
+      gap="small"
+      onDone={() => setShowContent(false)}
+      {...(!controlled ? rest : {})}
+    >
+      {!drop && (
+        <Header>
+          <Heading margin="none" level={2} size="small">
+            {heading ||
+              format({
+                id: 'dataFilters.heading',
+                messages: messages?.dataFilters,
+              })}
+          </Heading>
+          {layer && (
+            <Button
+              icon={<Close />}
```

Comment:
should we use `formClose` instead 
this looks big to me compared to the label
<img width="691" alt="Screen Shot 2022-12-16 at 1 47 09 PM" src="https://user-images.githubusercontent.com/42451602/208186256-ad2cdd2b-ca91-41c9-b938-cb8a780404ed.png">
<img width="500" alt="Screen Shot 2022-12-16 at 1 47 26 PM" src="https://user-images.githubusercontent.com/42451602/208186296-d65b2b4f-e810-4630-9a16-76a95d465830.png">



<img width="516" alt="Screen Shot 2022-12-16 at 1 46 11 PM" src="https://user-images.githubusercontent.com/42451602/208186122-030724e1-cd87-4d86-8912-815d4de1fcd9.png">

Reply:
Agree that FormClose would align with our later guidance.

Reply:
Remove `layer` for now.